File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    var reg = ['aya', 'ye', 'woo', 'ma']; // 가능한 발음
    var answer = 0;
    
    for (let word of babbling) {
        let prev = '';
        let temp = word;
        
        // 같은 발음이 2번 연속되는지 확인
        for (let r of reg) {
            if (temp.includes(r + r)) {
                // 연속된 발음이 있으면 종료
                prev = null;
                break;
            }
        }
        
        if (prev !== null) {
            // 정규 표현식에 일치하면 빈 문자열로 치환
            let str = temp.replace(/aya|ye|woo|ma/g, '');
            
            if (str === '') answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] reg = {"aya", "ye", "woo", "ma"};
        
        for (int i = 0; i < babbling.length; i++) {
            String prev = "";
            String temp = babbling[i];
            
            // 같은 발음이 2번 연속되는지 확인
            for (int j = 0; j < reg.length; j++) {
                if (temp.contains(reg[j] + reg[j])) {
                    // 연속된 발음이 있으면 종료
                    prev = null;
                    break;
                }
            }
            
            if (prev != null) {
                // 정규 표현식에 일치하면 빈 문자열로 치환
                String str = temp.replaceAll("aya|ye|woo|ma", "");
                
                if (str.equals("")) answer++;
            }
        }
        return answer;
    }
}